CodingStandard.md

```diff
@@ -0,0 +1,1425 @@
+should be ordered logically to help the reader.
+
+The #include for the class header should be the first substantive line in the
+source file, following the file header and the copyright block.
+
+Source code should be wrapped at 90 characters.
+
+
+**Inlining**
+
+Trivial member functions can be defined directly inside the class definition
+and do not require header comments.  They should not be marked `inline` as they
+are considered inline by default.
+
+Other simple member functions can be defined in a header file, but they should
+be defined outside the class definition, with header comments.  They should be
+marked `inline`.  This does not guarantee that the code will be inlined by the
+compiler, but rather it prevents violations of the "One Definition Rule" and
+link-time errors.
+
+Template functions defined in a header file should not be marked `inline`
+as they are considered inline by default.
+
+**Special Member Functions**
+
+The special member functions of a class (constructors, copy/move constructors,
+destructors, and copy/move assignment operators) should be declared in the
+class definition and defined below the class definition in the header file if
+they are inline, or in the source file otherwise.
+
+Undesired special member functions should be marked `= delete;` inside the
+public section of the class definition (rather than making them `private`).
+
+Except for simple struct types, default versions of all special member functions
+should be explicitly declared and defined.  Declare them in the class definition
+and define them using `= default;` in the source file.
+
+For example:
+
+```c++
+/// a simple struct type can implictly define special
+/// member functions and use default member initialization
+
+/*
+ * (struct header)
+ */
+struct Point
+{
+    double x{1.0};
+    double y{1.0};
+}
+
+/// class defined in .h file
+
+/*
+ * (class header)
+ */
+class Position
+{
+public:
+    Position() = delete;
+    Position(int lat_deg, int lat_min, int lat_sec,
+             int lon_deg, int lon_min, int lon_sec);
+    double distance_from(const Position& position);
+    ~Position();
+
+private:
+    double lat_;
+    double lon_;
+}
+
+/// methods are inline and defined in .h file
+/// or non-inline and defined in .cpp file
+
+/*
+ * (routine header)
+ */
+inline Position::Position(int lat_deg, int lat_min, int lat_sec,
+                          int lon_deg, int lon_min, int lon_sec)
+{
+    // ...
+}
+/*
+ * (routine header)
+ */
+inline Position::~Position() = default;
+/*
+ * (routine header)
+ */
+inline double Position::distance_from()
+{
+    // ...
+}
+```
+
+Single argument constructors and conversion functions should be marked
+`explicit` unless they are intended to participate in implicit conversions.  In
+that case, provide rationale in the conversion function routine header.
+
+**Variable Initialization**
+
+Prefer C++11 uniform initialization syntax (curly braces) for initializing 
+variables.
```

Comment:
Is using curly braces really the standard these days? That syntax bothers me a little. What is the issue with using constructors explicitly?

Maybe it is just me using languages other than C++ lately, but I would prefer something like:
`auto foo = Foo();` or `auto foo = Foo(1,2);`

Also, is something like `int x = 1;`fine, or does it have to be `int x = {1};` or `int x {1};`?

Reply:
Brace-initialization prohibits the compiler to perform implicit conversions to match constructor argument types.  <a href="https://stackoverflow.com/questions/18222926/why-is-list-initialization-using-curly-braces-better-than-the-alternatives">See this SO question for details.</a>